### ISSUE TITLE
Running tasks based on different OSes

### ISSUE THREAD

**Issue by sujitawake** [casey/just#1288]
I have a usecase where I need to install around 5+ binaries based on whatever `os()` built-in returns in the `justfile`. However, I see an error message that I am failing to debug. I also called the `myos` variable but no luck so far.

Any idea where I am doing it wrong?

### justfile

```
# Load environment variables from ".env"
set dotenv-load

myos := if os() == "macos" {
    "macos"
} else if os() == "linux" {
    "linux"
} else if os() == "windows" {
    "windows"
} else {
    "null"
}

# Install tools
tools_common:
    test := if os() == "macos" { `wget -q $BINURL_MAC_RG` } else if os() == "linux" { `wget -q $BINURL_LIN_RG` } else if os() == "windows" { `wget -q $BINURL_WIN_RG` } else { @echo "null" }
```

### .env

```
BINURL_LIN_RG='https://github.com/myrepo/package.x86_64-unknown-linux-musl.tar.gz'
BINURL_MAC_RG='https://github.com/myrepo/package.x86_64-apple-darwin.tar.gz'
BINURL_WIN_RG='https://github.com/myrepo/package.windows-x86_64.tar.gz'
```

When executing, I am seeing an error msg like the following:

```
test := if os() == "macos" { `eget -q $BINURL_MAC_RG` } else if os() == "linux" { `eget -q $BINURL_LIN_RG` } else if os() == "windows" { `eget -q $BINURL_WIN_RG` } else { @echo "null" }
sh: -c: line 0: syntax error near unexpected token `('
sh: -c: line 0: `test := if os() == "macos" { `eget -q $BINURL_MAC_RG` } else if os() == "linux" { `eget -q $BINURL_LIN_RG` } else if os() == "windows" { `eget -q $BINURL_WIN_RG` } else { @echo "null" }'
error: Recipe `tools_common` failed on line 16 with exit code 2
```

**Comment by sujitawake** [comment]
Also, I need to call wget multiple times (not sure how to do it) for different binary paths but in the above example, I am passing only one URL.

**Comment by runeimp** [comment]
You can't use Just's if statements _raw_ in recipes. And `:=` won't work either. So there are a few things to be aware of with Just.

1. Recipe lines are executed using the Bourne shell `sh` found on the system by default
2. Recipes that declare a shebang on the first line, `#!/bin/sh` or similar, execute the entire contents of a recipe as a single script file for the script interpreter (Bash shell, Python, Rexx, etc.)
3. Just's if statements must be wrapped in double-curly braces to work like so: `{{ IF_LOGIC }}` note the extra spaces around the IF_LOGIC

So you Justfile should probably look more like

```
# Load environment variables from ".env"
set dotenv-load

myos := if os() == "macos" {
    "macos"
} else if os() == "linux" {
    "linux"
} else if os() == "windows" {
    "windows"
} else {
    "null"
}

# Install tools
tools_common:
    test={{ if os() == "macos" { `wget -q $BINURL_MAC_RG` } else if os() == "linux" { `wget -q $BINURL_LIN_RG` } else if os() == "windows" { `wget -q $BINURL_WIN_RG` } else { @echo "null" } }}
```

Note that I've not tested this but it should be closer to what you need.

**Comment by casey** [comment]
Check out #1387! It adds `[linux]`, `[macos]`, `[windows]`, and `[unix]` attributes, which can be used to write different recipes with the same name that work on multiple OSes.

**Comment by casey** [comment]
It's basically done, all I need to do is fix the tests and add docs.

**Comment by casey** [comment]
Done in #1387 and released in 1.8.0. Doing different things on different operating systems has been pretty painful up until now, so I hope this helps!
